issue-001/solution.js

```diff
@@ -1,3 +1,38 @@
 function dataTypeCheck(a) {
     // implement your solution here
-}
\ No newline at end of file
+    type = typeof(a)
+    if (type === 'number') {
+        if (parseFloat(a) == a) {
+    	    //check for integer datatypes
+    	    if ( a % 1 === 0 ){
+    		    return a / 2
+    	    }
+    	    //check for float datatypes
```

Comment:
I think this is okay now since the number will either be an `integer` or a `float`. However, it will make more sense if you used `parseInt(a, 10)` as opposed to `parseFloat`.